### PR TITLE
support mapUnderscoreToCamelCase config

### PR DESCRIPTION
**Pull request by 362228416**
support mapUnderscoreToCamelCase config

### CODE REVIEW

**Comment by 362228416** [comment]
please merge pull and publish to maven repository

File: .github/project.yml [hunk 1]
```diff
@@ -1,5 +1,5 @@
 name: Quarkiverse MyBatis Extension
 release:
   current-version: 0.0.2
-  next-version: 0.0.3-SNAPSHOT
+  next-version: 0.0.3
```
Comment (zhfeng) [inline .github/project.yml:4]: Please don't change the version here. And the quarkiverse bot will help us for releasing the new version.

File: deployment/pom.xml [hunk 2]
```diff
@@ -3,7 +3,7 @@
     <parent>
         <groupId>io.quarkiverse.mybatis</groupId>
         <artifactId>quarkus-mybatis-parent</artifactId>
-        <version>0.0.3-SNAPSHOT</version>
+        <version>0.0.3</version>
```
Comment (zhfeng) [inline deployment/pom.xml:6]: some as above.

File: integration-test/pom.xml [hunk 3]
```diff
@@ -3,7 +3,7 @@
     <parent>
         <groupId>io.quarkiverse.mybatis</groupId>
         <artifactId>quarkus-mybatis-parent</artifactId>
-        <version>0.0.3-SNAPSHOT</version>
+        <version>0.0.3</version>
```
Comment (zhfeng) [inline integration-test/pom.xml:6]: some as above.

Comment (zhfeng) [inline integration-test/pom.xml:6]: some as above.

File: pom.xml [hunk 4]
```diff
@@ -11,7 +11,7 @@
     <groupId>io.quarkiverse.mybatis</groupId>
     <artifactId>quarkus-mybatis-parent</artifactId>
     <name>Quarkus - Mybatis - Parent</name>
-    <version>0.0.3-SNAPSHOT</version>
+    <version>0.0.3</version>
```
Comment (zhfeng) [inline pom.xml:14]: same as above.

File: runtime/pom.xml [hunk 5]
```diff
@@ -3,7 +3,7 @@
     <parent>
         <groupId>io.quarkiverse.mybatis</groupId>
         <artifactId>quarkus-mybatis-parent</artifactId>
-        <version>0.0.3-SNAPSHOT</version>
+        <version>0.0.3</version>
```
Comment (zhfeng) [inline runtime/pom.xml:6]: same as above.

**Review by zhfeng — CHANGES_REQUESTED** [review]

**Comment by zhfeng** [comment]
Thansk @362228416 for these contribution !

Can I ask you to rasie an issue to describe how this option mapUnderscoreToCamelCase can be used ? Also it could be better to provide some test cases.

**Comment by 362228416** [comment]
application.properties
```
quarkus.mybatis.map-underscore-to-camel-case=true
```
User.java
```
public class User {
    String userId;
    // getter setter
}
```
UserMapper.java
```java
public interface UserMapper {
   @Select("select user_id from user")
   List<User> findAll();
}
```

user_id =>  userId

The underline will automatically turn into a hump

**Comment by 362228416** [comment]
This is just to add a configuration, it will not be turned off by default, it will not affect the function, please feel free to update

File: runtime/src/main/java/io/quarkiverse/mybatis/runtime/MyBatisRuntimeConfig.java [hunk 6]
```diff
@@ -31,4 +31,11 @@
      */
     @ConfigItem(name = "initial-sql")
     public Optional<String> initialSql;
+
+    /**
+     * MyBatis initial sql
+     */
```
Comment (zhfeng) [inline runtime/src/main/java/io/quarkiverse/mybatis/runtime/MyBatisRuntimeConfig.java:37]: Please change this comment to reflect the right option. Thanks !

**Review by zhfeng — CHANGES_REQUESTED** [review]

**Comment by zhfeng** [comment]
@362228416 OK, I get it. So this option is from the MyBatis, right ? Can you check my comment above ? I will merge this PR till all of the them are fixed. The remain things look good to me. Thanks again !

**Comment by 362228416** [comment]
yes

File: runtime/src/main/java/io/quarkiverse/mybatis/runtime/MyBatisRuntimeConfig.java (hunk 6 above)
Comment (362228416) [reply runtime/src/main/java/io/quarkiverse/mybatis/runtime/MyBatisRuntimeConfig.java:37]: yes

**Comment by zhfeng** [comment]
@362228416 Do you have any update on this PR ? If you fix the things I mentioned above, I can merge the PR and raise a release for the new version. Thanks a lot !

Anyway, if you don't work on this any more, I will raise a new PR to replace this one to add this option from MyBatis.

**Comment by zhfeng** [comment]
Hi @362228416 I merged your changes in a new PR https://github.com/quarkiverse/quarkus-mybatis/pull/26 and will release 0.0.3 soon. Please can you check this option works in the new version of quarkus-mybatis ? Thanks !

@all-contributors add @362228416 for code, please !

**Pull request closed without merging**